doc/guides/timers-in-node.md

```diff
@@ -0,0 +1,154 @@
+
+### "Right after this" Execution ~ *setImmediate()*
+
+`setImmediate()` allows you to execute code at the beginning of the next 
+event loop cycle. This code will execute *before* any timers or IO operations. 
+I like to think of this code execution as happening "right after this", meaning 
+any code following the `setImmediate()` function call will execute before the 
+`setImmediate()` function argument. Here's an example:
+
+```js
+let order = 'before immediate\n';
+
+setImmediate(() => {
+  order += 'executing immediate\n';
+});
+
+order += 'after immediate\n';
+
+console.log(order);
+```
+
+The above function passed to `setImmediate()` will execute after all runnable 
+code has executed, and the console output will be:
+
+```shell
+before immediate
+after immediate
+executing immediate
+```
+
+Note: `process.nextTick()` is very similar to `setImmediate()`. The two major 
+differences are that `process.nextTick()` will run *before* any immediates that 
+are set. The second is that `process.nextTick()` is non-clearable, meaning once 
+you've scheduled code to execute with `process.nextTick()` you cannot stop that 
+code, unlike with `setImmediate()`.
+
+### "Deja vu" Execution ~ *setInterval()*
+
+If there is a block of code that you want to execute multiple times, you can 
+use `setInterval()` to execute that code. `setInterval()` takes a function 
+argument that will run and infinite number of times with a given millisecond 
+delay. Just like `setTimeout()`, the delay cannot be guaranteed because of 
+operations that may hold on to the event loop, and therefore should be treated 
+as an approximate delay. See the below example:
+
+```js
+function intervalFunc () {
+  console.log('Cant stop me now!');
+}
+
+setInterval(intervalFunc, 1500);
+```
+In the above example, `intervalFunc()` will execute every 1500 milliseconds, or 
+1.5 seconds, until it is stopped (see below).
+
+## Master of the Timerverse
+
+What would a Code Time Machine be without the ability to turn it off? 
+`setTimeout()`, `setImmediate()`, and `setInterval()` return a timer object 
+that can be used to reference the set timeout, immediate, or interval object. 
+By passing said objective into the respective `clear` function, execution of 
+that object will be halted completely. The respective functions are 
+`clearTimeout()`, `clearImmediate()`, and `clearInterval()`. See the example 
+below for an example of each:
+
+```js
+let timeoutObj = setTimeout(() => {
+  console.log('timeout beyond time');
+}, 1500);
+
+let immediateObj = setImmediate(() => {
+  console.log('immediately executing immediate');
+});
+
+let intervalObj = setInterval(() => {
+  console.log('interviewing the interval');
+}, 500);
+
+clearTimeout(timeoutObj);
+clearImmediate(immediateObj);
+clearInterval(intervalObj);
+```
+
+## Last Train to Nowhere
+
+The node Timer API provides two functions intended to augment timer behavior 
+with `unref()` and `ref()`. If you have a timer object scheduled using a 
```

Comment:
Should specify that these are on `Timeout`s prototype. Also need to state above that intervals also give you a `Timeout` class, but immediates give you an `Immediate`.